scripts/common/proxy.sh

```diff
@@ -47,6 +47,46 @@ function proxy_bootstrap() {
     fi
 }
 
+# check_proxy_config tries to check if is possible connect with the registry
+# the following code will check if the proxy is invalid by doing a check as following
+# ~$ sudo HTTP_PROXY=http://77.89.35.50:8080 HTTPS_PROXY=http://77.89.35.50:8080 ctr image pull docker.io/library/nginx:latest
+# docker.io/library/nginx:latest: resolving      |--------------------------------------|
+# elapsed: 29.9s                  total:   0.0 B (0.0 B/s)
+# INFO[0030] trying next host                              error="failed to do request: Head \"https://registry-1.docker.io/v2/library/nginx/manifests/latest\": proxyconnect tcp: dial tcp 77.89.35.50:8080: i/o timeout" host=registry-1.docker.io
+# ctr: failed to resolve reference "docker.io/library/nginx:latest": failed to do request: Head "https://registry-1.docker.io/v2/library/nginx/manifests/latest": proxyconnect tcp: dial tcp 77.89.35.50:8080: i/o timeout
+function check_proxy_config() {
+    if [ -n "$CONTAINERD_VERSION" ]; then
+        logStep "Checking proxy configuration with Containerd"
+
+        # Echo containerd Proxy config:
+        local proxy_config_file="/etc/systemd/system/containerd.service.d/http-proxy.conf"
+        if [ -f "$proxy_config_file" ]; then
+            echo ""
+            log "Proxy config:"
+            grep -v -e '^\[Service\]' -e '^# Generated by kURL' "$proxy_config_file"
+            echo ""
+        else
+            log "Skipping test. No HTTP proxy configuration found."
+            return
+        fi
+
+        # Get proxy configuration from http-proxy.conf file
+        local http_proxy=$(grep -E '^\s*Environment\s*=\s*"HTTP_PROXY=' "$proxy_config_file" | sed -E 's/^\s*Environment\s*=\s*"HTTP_PROXY=(.*)"/\1/')
```

Comment:
Do we need to read these from the containerd configuration ? I mean, if containerd is configured to use proxy then we would not need to pass this in the `ctr` command line. What am I not seeing here ?

Reply:
Unfortunately, that's correct. If we simply use the ctr image pull command with an invalid configuration, it doesn't fail or raise any errors. This behavior suggests that when calling ctr directly, it doesn't utilize the settings that are configured for pulling images defined within a Deployment, for instance.